arduino/Remote_Prosthetic_Control.ino

```diff
@@ -0,0 +1,244 @@
+//Thursday, April 10th 2014      By: Cameron Clark
+
+////This sketch uses data from a infared rangefinder connected to analog pin A0
+//and uses that to operate a servo connected to digital Pin 9 that is aiming
+//the beam of a laser up so that a users hand can "pluck" the laser which
+//wobbles back to place.
+
+//General Setup:
+//This is designed as a state-based fuction. It is always in one of these states:
+//0: At rest.
+//1: Tracking a right-handed pluck.
+//2: Tracking a left-handed pluck.
+//3: Pluck right.
+//4: Pluck left.
+//5: Returning to rest while listening.
+
+
+
+//Additions planned: 
+//-Run and power a laser
+//-Send out serial info for use in Synth program
+
+#include <Servo.h>
+#include <Average.h>
+
+//Variables that might be adjusted.
+float decayLength = .1;
+int servoPin = 9;
+int delayAmount = 0;
+int pluckLength = 50;
+
+//Variables that should be constants.
+Servo myservo;
+int decayCounter = 0;
+float decay = 0;
+float rangeFinderX;
+float userAngle = 0;
+float initialAngle = 0;
+float servoAngle = 0;
+float lastLocationX = 0;
+int currentState = 0;
+int lastState = 0;
+float pluckStrength = 0;
+float handSpeed = 0;
+
+void setup() {
+  myservo.attach(servoPin);
+  Serial.begin(9600);
+  delay(1000);
+}
+
+
+
+
+void loop() {  
+  
+  rangeFinderX = rangeFinderValue();   //Get the rangefinder value.
+  
+  if (rangeFinderX == 132.00)
+  {
+    rangeFinderX = lastLocationX;
+  }
+ 
+  Serial.print("State before if's:  ");
+  Serial.print(currentState);
+  Serial.print(" RangeFinder Value before Ifs: ");
+  Serial.print(rangeFinderX);
+
+
+  
+  if (currentState == 0) {                     //State 0: At rest and listening...
+  
+      if (rangeFinderX >= 0 && rangeFinderX < 50)
+      {
```

Comment:
@CameronFoxly, Are we following a deterministic state approach here?

Maybe a switch case implementation can make it more understandable of the state flow, just a suggestion.

And, should the adjusted variables be passed through command lines `sys.arg` instead.  I believe it will give the user much more freedom to adjust them than everytime editing the variables ?

Let me know about that @CameronFoxly @Cleo123.

I will open a Pull Request for that :smiley: